sql/updates/README.md

```diff
@@ -1,9 +1,31 @@
-Folder for update SQL-scripts named pre-oldversion--newversion or post-oldversion--newversion
+## Extension updates
 
-pre files are run before reloading the functions. These should include new
-tables, constraints, indexes, etc.
+This directory contains SQL files (modfiles) with idempotent
+modifications that happen when updating from one version to another.
 
-post functions are run after reloading the functions. These include
-any entities depending on functions-- e.g. triggers.
+The actual update scripts are compiled from modfiles by concatenating
+them with the current source code. Modfiles should always be
+first. Update scripts can "jump" several versions by using multiple
+modfiles in order.
 
-It is assumed that there is only one file for pre or post matching the newversion.
+To ensure that this update process works, there are a few principles
+to considered.
+
+1. Modfiles should, generally, only contain `ALTER` or `DROP` commands
+   that change or remove objects.
+2. Modifications must be idempotent. In most cases that means commands
+   should have an `IF NOT EXISTS` clause, if possible. The reason is
+   that some modfiles might try to, e.g., DROP objects that aren't
+   present because they only exists in an intermediate version of the
+   database, which is skipped over.
+3. Modifications that depend on objects or functions that are NOT
```

Comment:
This is actually more complex. I'd say that all functions (whether or not they are present before the update) used in the modfiles should be defined in the modfile before use. The problem is that if the function did not exist in 0.1.0 you can never be sure it exists before the modfile is run.

Reply:
You can trust that functions that are in the release your are upgrading from exist. For instance, if I am upgrading from 0.6.0 to 0.9.0, all functions in 0.6.0 must be present.

Reply:
But you don't know which version a user is updating from. The uses can be updating from 0.6.0 or 0.1.0. So the modfiles cannot assume anything about the functions present in the previous version (this is a consequence of the fact that all functions are installed /after/ the modfiles are run).

Reply:
Yeah, that's true.